File: src/main/java/mountainrangepvp/game/world/ChatManager.java
package mountainrangepvp.game.world;

import java.util.Collections;
import java.util.LinkedList;
import java.util.List;

/**
 * @author lachlan
 */
public class ChatManager {

    private final List<ChatLine> lines;

    private boolean chatting;
    private String currentLine;

    public ChatManager() {
        lines = new LinkedList<>();
    }

    public List<ChatLine> getLinesHead(int length) {
        return Collections.unmodifiableList(lines.subList(0,
                                                          Math.min(lines.size(),
                                                                   length)));
    }

    public boolean isChatting() {
        return chatting;
    }

    public String getCurrentLine() {
        return currentLine;
    }
}

File: src/main/java/mountainrangepvp/game/world/ChatManager.scala
package mountainrangepvp.game.world

class ChatManager {
  private var lines: List[ChatLine] = List.empty
  private var chatting: Boolean = false
  private var currentLine: String = null

  def getLinesHead(length: Int): List[ChatLine] = lines.take(length)

  def isChatting = chatting

  def getCurrentLine = currentLine
}
